### PR TITLE
group methods based on public, ownerOnly, internal

### PR DESCRIPTION
**Pull request by kneelsdev**

### CODE REVIEW

File: contracts/Archetype.sol [hunk 1]
```diff
@@ -512,11 +503,50 @@ contract Archetype is ERC721A__Initializable, ERC721AUpgradeable, ERC721A__Ownab
...
+  //
+  function _startTokenId() internal view virtual override returns (uint256) {
+    return 1;
+  }
+
+  function computePrice(
+    uint128 price,
+    uint256 numTokens,
+    bool affiliateUsed
+  ) internal view returns (uint256) {
```
Comment (kneelsdev) [inline contracts/Archetype.sol:537]: was there a reason this was public? I moved it to internal. Can change it back.

Comment (kneelsdev) [reply contracts/Archetype.sol:537]: @coindegen ^

File: test/archetype.test.ts [hunk 2]
```diff
@@ -935,20 +935,43 @@ describe("Factory", function () {
...
+    const newCollection = await factory.createCollection(owner.address, DEFAULT_NAME, DEFAULT_SYMBOL, DEFAULT_CONFIG);
+    const result = await newCollection.wait();
+    const newCollectionAddress = result.events[0].address || "";
+    const NFT = await ethers.getContractFactory("Archetype");
+    const nft = NFT.attach(newCollectionAddress);
+
+    await expect(nft.connect(owner).setSuperAffiliatePayout(minter.address)).to.be.revertedWith("caller is not the platform");
+    await nft.connect(platform).setSuperAffiliatePayout(minter.address);
+
+    await expect((await nft.connect(minter).config()).superAffiliatePayout).to.be.equal(minter.address);
```
Comment (coindegen) [inline test/archetype.test.ts:974]: Can we add a test for not exceeding max supply. 🙏 

Has to account for burned tokens as well.

File: contracts/Archetype.sol [hunk 3]
```diff
@@ -98,29 +98,38 @@ contract Archetype is ERC721A__Initializable, ERC721AUpgradeable, ERC721A__Ownab
     uint128 platform;
   }
```
Comment (coindegen) [inline contracts/Archetype.sol:100]: formatting is off in your IDE. I use the VS Code Solidity extension

File: contracts/Archetype.sol [hunk 4]
```diff
@@ -98,29 +98,38 @@ contract Archetype is ERC721A__Initializable, ERC721AUpgradeable, ERC721A__Ownab
...
 
+  struct BurnConfig {
```
Comment (coindegen) [inline contracts/Archetype.sol:101]: also bump the version to 0.3.3

File: test/archetype.test.ts [hunk 5]
```diff
@@ -936,19 +938,193 @@ describe("Factory", function () {
...
+
+    await expect((await nft.connect(minter).config()).superAffiliatePayout).to.be.equal(
+      minter.address
+    );
+  });
+
+  it("test max supply checks", async function () {
+    const [accountZero, accountOne] = await ethers.getSigners();
+    DEFAULT_CONFIG["maxBatchSize"] = 5000;
+    DEFAULT_CONFIG["maxSupply"] = 5000;
```
Comment (coindegen) [inline test/archetype.test.ts:990]: use dot notation instead, eg DEFAULT_CONFIG.maxBatchSize

**Pull request merged**